Xamarin.Forms.Platform.iOS/Renderers/PickerRenderer.cs

```diff
@@ -138,13 +139,51 @@ void UpdateTextColor()
 				Control.TextColor = textColor.ToUIColor();
 		}
 
+		protected override void Dispose(bool disposing)
+		{
+			if (_disposed)
+				return;
+
+			_disposed = true;
+
+			if (disposing)
+			{
+				_defaultTextColor = null;
+
+				if (_picker != null)
+				{
+					if (_picker.Model != null)
+					{
+						_picker.Model.Dispose();
+						_picker.Model = null;
+					}
+
+					_picker.RemoveFromSuperview();
+					_picker.Dispose();
+					_picker = null;
+				}
+
+				if (Control != null)
+				{
+					Control.EditingDidBegin -= OnStarted;
+					Control.EditingDidEnd -= OnEnded;
+				}
+
+				if(Element != null)
+					((INotifyCollectionChanged)Element.Items).CollectionChanged -= RowsCollectionChanged;
```

Comment:
@StephaneDelcroix I'm not 100% sure about this one. In a working situation, `oldElement` and `newElement` should be changed in a balanced way. When the object is being disposed, `newElement` should be set to null and `OnElementChanged` should be fired. This should unregister the handler because `oldElement` is not null.

This is just extra measure in case things don't work. Perhaps it makes sense to not do this and potentially end up with a leak so it could be investigated further and the root cause can be fixed.